### ISSUE TITLE
Undetected unreachable code

### ISSUE THREAD

**Issue by falsandtru** [microsoft/TypeScript#35835]
These are obvious mistakes.

**TypeScript Version:**  3.7.x-dev.20191221

**Search Terms:**

**Code**

```ts
declare const a: boolean;
(() => {
    if (a) return;
    if (a) {
        return; // Unreachable
    }
});
while (1) {
    if (a) break;
    if (a) {
        break; // Unreachable
    }
}
```

**Expected behavior:**
Detect unreachable code.
**Actual behavior:**
not detected.
**Playground Link:** http://www.typescriptlang.org/play/index.html?ts=3.8.0-dev.20191221#code/CYUwxgNghgTiAEYD2A7AzgF3lAXPARkkhCFCgNwCwAUABS0CU8AvAHzwDeN8P8AlgDN4tKEzgYArjArdeg4aM6zevcVIrwA9JvgBVFHChgAFlHwll8AL40rDKtQDuxviWEBGJl2or5IpviGANYOvkL+Sj4qvIGkIVo6+oYmZhZRPDbUVkA

**Related Issues:**

**Comment by jcalz** [comment]
Related: #17358

**Comment by falsandtru** [comment]
The definitive difference is other similar issues describe the place where is unreachable at type-level but actually reachable at runtime but this issue describes the place where is unreachable even at runtime.

**Comment by falsandtru** [comment]
And type-level unreachable clauses of other examples must be allowed to handle exceptional cases at runtime but this is always just a mistake.
